docs/site-generator/hugo-template/Makefile

```diff
@@ -0,0 +1,78 @@
+# Default shell
+SHELL := /bin/bash
+
+# Config
+DESTINATION ?= public
+LANDING_PATH ?=
+
+# System
+url_os = 64bit
+os = amd64
+ifeq ($(os),Windows_NT)
+    arch = windows
+    url_arch = Windows
+else
+    uname_s := $(shell uname -s)
+    ifeq ($(uname_s),Linux)
+			arch = linux
+			url_arch = Linux
+    endif
+    ifeq ($(uname_s),Darwin)
+			arch = darwin
+			url_arch = MacOS
+    endif
+endif
+
+# Hugo variables
+hugo_version := 0.21
+hugo_path := $(shell pwd)/.hugo
+hugo_url = github.com/spf13/hugo
+hugo_url_name := hugo_$(hugo_version)_$(url_arch)-$(url_os)
+
+# Tools
+curl = curl -L
+hugo = $(hugo_path)/hugo
+mkdir = mkdir -p
+npm = npm
+untar = tar -xf
+rename = mv
+remove = rm -rf
+
+## Lists all recipes
+list:
+	@grep '^##' Makefile -A 1
+
+## Install dependencies
+dependencies: hugo-dependencies import-commons npm-dependencies
+
+## Builds hugo project compiling assets and hugo files
+hugo-build:
+	$(npm) run build
+	$(hugo) --config=hugo.config.yaml --destination=$(DESTINATION)
+
+# Updates hugo dependencies
+hugo-dependencies:
+	@if [[ ! -f $(hugo) ]]; then \
+		$(mkdir) $(hugo_path); \
+		cd $(hugo_path); \
+		ext="zip"; \
+		if [ "$(arch)" == "linux" ]; then ext="tar.gz"; fi; \
+		file="hugo.$${ext}"; \
+		$(curl) https://$(hugo_url)/releases/download/v$(hugo_version)/$(hugo_url_name).$${ext} -o $${file}; \
+		if [ "$(arch)" == "linux" ]; then tar -xvzf $${file}; else unzip $${file}; fi; \
+	fi;
+
+export_filename = $(shell pwd)/landing-common.tar
+file_list_name = .list.gitignore
+# Imports common stuff
+import-commons:
+	$(MAKE) -C $(LANDING_PATH) export-landing-commons \
```

Comment:
The dependencies of the generation of a _landing-like_ site should be controlled from the _landing-like-generator_
Otherwise it could be out of sync with landing templates.

To ensure that all the dependencies are fetched only once and cached for the next time, it can be called `make template-dependencies` from the `init.sh` as done at [docs/init.d/init.sh#41](https://github.com/src-d/ci/pull/23/files#diff-670ec070448b5ca1f50c55b6543f17c2R41)